src/internal/_arrayChainRec.js

```diff
@@ -0,0 +1,19 @@
+module.exports = (function() {
+  function stepNext(x) { return {value: x, done: false }; }
+  function stepDone(x) { return {value: x, done: true }; }
+
+  return function _chainRec(f, i) {
```

Comment:
Could we use `x` here rather than `i`? `i` makes me think "index" or "integer", neither of which applies here.